bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/discovery/VehicleDiscovery.java

```diff
@@ -174,50 +79,150 @@ public void setThingHandler(ThingHandler handler) {
 
     @Override
     protected void startScan() {
-        bridgeHandler.ifPresent(MyBMWBridgeHandler::requestVehicles);
+        logger.trace("xxxVehicleDiscovery.startScan");
+        discoverVehicles();
     }
 
     @Override
     public void deactivate() {
+        logger.trace("xxxVehicleDiscovery.deactivate");
+
         super.deactivate();
     }
 
-    public static String getServices(Vehicle vehicle, String suffix, boolean enabled) {
-        StringBuffer sb = new StringBuffer();
-        List<String> l = getObject(vehicle.capabilities, enabled);
-        for (String capEntry : l) {
-            // remove "is" prefix
-            String cut = capEntry.substring(2);
-            if (cut.endsWith(suffix)) {
-                if (sb.length() > 0) {
-                    sb.append(Constants.SEMICOLON);
+    public void discoverVehicles() {
+        logger.trace("xxxVehicleDiscovery.discoverVehicles");
+
+        myBMWProxy = bridgeHandler.get().getMyBmwProxy();
+
+        try {
+            Optional<List<@NonNull Vehicle>> vehicleList = myBMWProxy.map(prox -> {
+                try {
+                    return prox.requestVehicles();
+                } catch (NetworkException e) {
+                    throw new IllegalStateException("vehicles could not be discovered: " + e.getMessage(), e);
```

Comment:
so the NetworkException is a checked exception. I have to catch it inside of the Lambda expression and to avoid that I have an uncaught exception I catch it outside of the Lambda.
I tried some different things and this was the solution I ended up which was working fine...

Reply:
Let's leave it like this.